### DISCUSSION TITLE
Dataset Suitability Analysis

### DISCUSSION THREAD

**Discussion by amqdn** [fourthbrain-students/glg-ack#3] [General]
Hi,

I'm just going through datasets and jotting down notes on why I think it might work for our purposes and why it might not. I'm going through the [Hugging Face Datasets](https://huggingface.co/datasets) library for now, but I may go through other sources. All of the datasets in this post are found by following that link and filtering by: `question-answering` `en` `monolingual`

- [x] [adversarial_qa](https://huggingface.co/datasets/viewer/?dataset=adversarial_qa): Relevant technical queries; finding the topic hierarchy will make this easier to annotate
- [ ] [ambig_qa](https://huggingface.co/datasets/viewer/?dataset=ambig_qa): Grade-school level
- [ ] [babi_qa](https://huggingface.co/datasets/viewer/?dataset=babi_qa): Logical story-telling; not specifically technical
- [ ] [codah](https://huggingface.co/datasets/viewer/?dataset=codah): Not really a QA dataset; more like entailment and logic
- [x] [covid_qa_deepset](https://huggingface.co/datasets/viewer/?dataset=covid_qa_deepset): The questions here all have a medical context
- [ ] [cuad](https://huggingface.co/datasets/viewer/?dataset=cuad): Legal contracts, but the setup of the data isn't easily massaged
- [ ] [doc2dial](https://huggingface.co/datasets/viewer/?dataset=doc2dial): Unclear if domains overlap our objective
- [ ] [drop](https://huggingface.co/datasets/viewer/?dataset=drop): Hard to annotate; unclear if domains overlap
- [x] [eli5](https://huggingface.co/datasets/viewer/?dataset=eli5): Good questions; moderately hard to annotate
- [ ] [freebase_qa](https://huggingface.co/datasets/viewer/?dataset=freebase_qa): Hard to annotate
- [ ] [gooaq](https://huggingface.co/datasets/viewer/?dataset=gooaq): Too noisy
- [x] [grail_qa](https://huggingface.co/datasets/viewer/?dataset=grail_qa): Very good dataset, IMO. Annotated in just the way we want
- [x] [head_qa](https://huggingface.co/datasets/viewer/?dataset=head_qa): Exam questions for medical specialists; the English portion should do
- [ ] [mc_taco](https://huggingface.co/datasets/viewer/?dataset=mc_taco): Hard to annotate; no clear overlap
- [x] [med_hop](https://huggingface.co/datasets/viewer/?dataset=med_hop): Not sure if worth it. The queries are not useful, but the text passages are highly medically technical
- [x] [medical_dialog](https://huggingface.co/datasets/viewer/?dataset=medical_dialog): Can't preview. Possibly useful
- [ ] [mocha](https://huggingface.co/datasets/viewer/?dataset=mocha): General reasoning dataset; eh
- [ ] [mrqa](https://huggingface.co/datasets/viewer/?dataset=mrqa): Hard to annotate
- [ ] [msr_sqa](https://huggingface.co/datasets/viewer/?dataset=msr_sqa): Specifically queries on tabular data; no clear overlap or annotation route
- [ ] [multi_re_qa](https://huggingface.co/datasets/viewer/?dataset=multi_re_qa): No text data, only spans
- [ ] [narrativeqa](https://huggingface.co/datasets/viewer/?dataset=narrativeqa): No clear overlap or annotation scheme
- [x] [neural_code_search](https://huggingface.co/datasets/viewer/?dataset=neural_code_search): Pretty good overlap for technology
- [ ] [newsqa](https://github.com/Maluuba/newsqa): No clear overlap or annotation scheme
- [ ] [nq_open](https://huggingface.co/datasets/viewer/?dataset=nq_open): Open-domain questions; hard to annotate
- [ ] [piqa](https://huggingface.co/datasets/viewer/?dataset=piqa): Physical activity queries; no clear overlap or annotation scheme
- [ ] [proto_qa](https://huggingface.co/datasets/viewer/?dataset=proto_qa): No clear overlap or annotation scheme
- [x] [pubmed_qa](https://huggingface.co/datasets/viewer/?dataset=pubmed_qa): Very relevant medical queries
- [ ] [qa_srl](https://huggingface.co/datasets/viewer/?dataset=qa_srl): No clear overlap or annotation scheme
- [ ] [qed](https://huggingface.co/datasets/viewer/?dataset=qed): No clear overlap or annotation scheme
- [ ] [quac](https://huggingface.co/datasets/viewer/?dataset=quac): Open-domain questions; hard to annotate, unclear overlap
- [ ] [ropes](https://huggingface.co/datasets/viewer/?dataset=ropes): Somewhat relevant text, but hard to annotate
- [ ] [selqa](https://huggingface.co/datasets/viewer/?dataset=selqa): Potentially good queries, dependent on filtering `section` column
- [x] [sharc](https://huggingface.co/datasets/viewer/?dataset=sharc): Sufficiently technical queries, but requires a fair bit of massaging
- [ ] [simple_questions_v2](https://huggingface.co/datasets/viewer/?dataset=simple_questions_v2): Open-domain; hard to annotate
- [ ] [squad](https://huggingface.co/datasets/viewer/?dataset=squad): Open-domain; no clear overlap
- [ ] [wiki_hop](https://huggingface.co/datasets/viewer/?dataset=wiki_hop): Open-domain; strange formatting
- [x] [yahoo_answers_qa](https://huggingface.co/datasets/viewer/?dataset=yahoo_answers_qa): Short queries, but already annotated
- [x] [zest](https://huggingface.co/datasets/viewer/?dataset=zest): Potentially relevant queries, dependent on `domain` column
- [ ] [dynabench/qa](https://huggingface.co/datasets/viewer/?dataset=dynabench/qa): Weird format; hard to annotate

**Comment by chuckcurtis** [comment]
Only glanced at this briefly and not sure we would end up using the data, but seems similar to some ideas we've had and could be useful as a reference if nothing else:

https://github.com/duyet/skill2vec-dataset

**Reply by amqdn** [reply]
Yes. Looks like we could at least use it to think about NER and as an example for the types of terms we might be tracking. 

The dataset Kinga mentioned includes `IT Jobs`, which we can probably massage into `Technology` examples: https://www.kaggle.com/airiddha/trainrev1

**Reply by kinga-k-farkas** [reply]
I am not sure what we are looking, here is loads of data by topic:
https://archive.org/download/stackexchange

**Reply by kinga-k-farkas** [reply]
Here is some medical data:  https://github.com/socd06/medical-nlp
